src/Functions/array/arrayAUC.cpp

```diff
@@ -197,14 +221,22 @@ class FunctionArrayAUC : public IFunction
         if (!col_array1->hasEqualOffsets(*col_array2))
             throw Exception(ErrorCodes::BAD_ARGUMENTS, "Array arguments for function {} must have equal sizes", getName());
 
+        // Handle third argument for scale (if passed, otherwise default to true)
+        bool scale = true;
+        if (number_of_arguments == 3)
+        {
+            scale = arguments[2].column->getBool(0); // Assumes it's a scalar boolean column
```

Comment:
Arguments in ClickHouse SQL functions can be constant or non-constant.

For example, in `SELECT someFunc(1, a) FROM tab`, `1` is a const argument and `a` is a non-const argument. There is also `materialize`, e.g. `SELECT someFunc(materialize(1), a) FROM tab` which transforms a const into a non-const argument.

In l. 228, we assume that the argument is const (only the value in the 1st row is read). When it is non-const, then the behavior will be surprising. It makes sense to disallow non-const scale arguments in the first place. This can be done in `getReturnTypeImpl` via `isColumnConst`, see e.g. src/Functions/randomFixedString.cpp for an example.

Reply:
i tried following the example you said, but I only find isColumnConst being used with ColumnWithTypeAndName

when I try to do it with DataTypes, the build fails with

```
00:31: error: no member named 'column' in 'std::shared_ptr<const DB::IDataType>'
  200 |             if (!arguments[0].column || !isColumnConst(*arguments[0].column))
      |                  ~~~~~~~~~~~~ ^
```

Reply:
Replace

```cpp
DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
```

by

```cpp
DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
```

and it will work.

Reply:
Now it built, just pushed this change

But I have a doubt

SELECT arrayAUC([0, 1, 1], [0, 0, 1], materialize(false)); -- { serverError ILLEGAL_COLUMN }

As I understood this should break now, right? But it isn't, it is working like before

Reply:
Yes, that should throw an error now ... hmm.

Reply:
I will try to investigate later

Reply:
Nevermind, I quickly made the needed fixes by myself.

Reply:
thank you!